### PR TITLE
Strip public keyword from managedToNative stub

### PR DESCRIPTION
**Pull request by jtschuster**
Accessibility modifiers on an interface method would be copied from the interface to the generated explicit implementations. Accessibility modifiers aren't allowed on explicit implementations, so the compilation would fail. This PR strips accessibility modifiers from the IncrementalMethodStubGenerationContext data.

Protected and Private interface methods are not supported and not tested.

### CODE REVIEW

**Review by jkoritzinsky — COMMENTED** [review]
Can you file an issue for us to cleanly emit a diagnostic for methods with `protected` modifiers?

File: src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SyntaxExtensions.cs [hunk 1]
```diff
@@ -81,6 +81,20 @@ public static SyntaxTokenList StripTriviaFromTokens(this SyntaxTokenList tokenLi
...
+            SyntaxToken[] strippedTokens = new SyntaxToken[tokenList.Count];
+            for (int i = 0; i < tokenList.Count; i++)
+            {
+                if (tokenList[i].Kind() is SyntaxKind.PublicKeyword or SyntaxKind.InternalKeyword or SyntaxKind.ProtectedKeyword or SyntaxKind.PrivateKeyword)
+                {
+                    continue;
+                }
+                strippedTokens[i] = tokenList[i];
+            }
+            return new SyntaxTokenList(strippedTokens);
```
Comment (jkoritzinsky) [inline src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SyntaxExtensions.cs:95]: nit: I'd rather do a `List<SytnaxToken>` here than an array so we don't have empty tokens we're carrying around.

**Review by jkoritzinsky — APPROVED** [review]

**Comment by jtschuster** [comment]
> Can you file an issue for us to cleanly emit a diagnostic for methods with `protected` modifiers?

Filed https://github.com/dotnet/runtime/issues/84499

**Comment by jtschuster** [comment]
Test failures are known

**Pull request merged**